internal/server/middleware.go

```diff
@@ -73,7 +73,9 @@ func handleInfraDestinationHeader(c *gin.Context) error {
 // possibly also of the destination.
 func authenticatedMiddleware(srv *Server) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		withDBTxn(c.Request.Context(), srv.DB(), func(tx *gorm.DB) {
+		withDBTxn(c.Request.Context(), srv.DB().GormDB(), func(db *gorm.DB) {
+
+			tx := data.NewTransaction(db, 0)
```

Comment:
As it was before, we now set the orgID in the transaction as part of these middleware. But until we have an org identified we query with orgID =0, which should ensure we never match an org.